packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/TileOverlayController.java

```diff
@@ -25,14 +23,8 @@ void clearTileCache() {
     tileOverlay.clearTileCache();
   }
 
-  Map<String, Object> getTileOverlayInfo() {
-    Map<String, Object> tileOverlayInfo = new HashMap<>();
-    tileOverlayInfo.put("fadeIn", tileOverlay.getFadeIn());
-    tileOverlayInfo.put("transparency", tileOverlay.getTransparency());
-    tileOverlayInfo.put("id", tileOverlay.getId());
-    tileOverlayInfo.put("zIndex", tileOverlay.getZIndex());
-    tileOverlayInfo.put("visible", tileOverlay.isVisible());
-    return tileOverlayInfo;
+  TileOverlay getTileOverlay() {
+    return tileOverlay;
```

Comment:
I could keep the old structure and have this convert to a Pigeon data structure here if you prefer, it just seemed weird to me to have serialization details several layers removed from the layer that is defining the serialization API. Returning the object and then consolidating the serialization in one place seemed cleaner to me.